redbpf/src/load/loader.rs

```diff
@@ -59,6 +60,19 @@ impl Loader {
             }
         }
 
+        // bpf_map_type_BPF_MAP_TYPE_RINGBUF
+        for m in module.maps.iter_mut().filter(|m| m.kind == 27) {
+            let name = m.name.clone();
+            let map = RingBufMap::bind(m).unwrap();
+            let stream = RingBufMessageStream::new(name.clone(), map);
+            let mut s = sender.clone();
+            let fut = stream.for_each(move |events| {
+                s.start_send((name.clone(), events)).unwrap();
+                future::ready(())
+            });
```

Comment:
I've used this branch to write some production-ish code and I have some feedback.

This works like a charm but I think the API here is a bit restrictive because there's no easy way to listen on ringbufs individually. Example case: I want to start my own async task for each ringbuf to forward incoming events to a TCP connection. With this approach, I'll have to filter all events to look for particular ringbufs that interest me, then likely forward these events through an mpsc queue, etc. etc.

Instead, I propose to store ringbufs in a separate `HashMap`:

```suggestion
            let mut ringbufs = HashMap::new();
            ...
            ringbufs.insert(name, stream);
```
and make it publicly accessible from `struct Loaded`, e.g.:

```rust
pub struct Loaded {
    pub events: mpsc::UnboundedReceiver<(String, <PerfMessageStream as Stream>::Item)>,
    //...
    pub ringbufs: HashMap<String, RingBufMessageStream>,
}
```

This would also simplify the implementation.

What do you think? cc @rhdxmr 

@yobiscus and thanks for working on this, it's an awesome feature! :)

Reply:
@nbaksalyar that makes a lot of sense.

I'm afraid this PR is a bit stale, so at this stage we might need another revision, + we can't merge without DCO from @yobiscus . Hang in there for another week or so to see if they re-appear, and we can figure out a way forward if not.

Reply:
Thanks for the much needed poke @nbaksalyar :) I am currently updating my environment so that it is compatible with the latest redbpf changes. I have rebased my changes against the main branch and hope to push something functional tomorrow.

Unfortunately I have not found time to address @rhdxmr comments above, specifically the comment about pointer alignment. There's a bit of a learning curve needed to address this which I don't currently have bandwidth for. Any help closing on these issues after I rebase my branch is very much appreciated!

Reply:
Pushed and signed-off. Sorry I wasn't able to implement your suggestion @nbaksalyar. Since this content is feature gated, would it be okay to merge this as is and address the comments after-the-fact, even if the changes are not backwards compatible?

Reply:
Sure, no worries! I can help with making this change if you'd like.

Reply:
Thanks everyone, this looks good, once the CI passes I can merge this!